test/dynamo/test_misc.py

```diff
@@ -5857,6 +5857,31 @@ def f():
         self.assertEqual(f(), torch.zeros(2, 2))
         self.assertEqual(opt_f(), torch.ones(2, 2))
 
+    def test_is_fullgraph_compiling(self):
+        def f():
+            if torch._dynamo.is_fullgraph_compiling():
+                return torch.ones(2, 2)
+            else:
+                return torch.zeros(2, 2)
+
+        def g():
+            if torch._dynamo.is_fullgraph_compiling():
+                return torch.ones(2, 2)
+            else:
+                return torch.zeros(2, 2)
+
+        self.assertEqual(f(), torch.zeros(2, 2))
+
+        opt_f = torch.compile(f, fullgraph=True)
+
+        self.assertEqual(f(), torch.zeros(2, 2))
+        self.assertEqual(opt_f(), torch.ones(2, 2))
+
+        opt_g = torch.compile(g, fullgraph=False)
+
+        self.assertEqual(g(), torch.zeros(2, 2))
+        self.assertEqual(opt_g(), torch.zeros(2, 2))
```

Comment:
One issue here is that if calling twice the same:

```python
opt_f = torch.compile(f, fullgraph=False)
f()
opt_f = torch.compile(f, fullgraph=True)
f()
```

somehow `torch.compile` does not initialize new `InstructionTranslator` at the second call, so the `one_graph` attribute is wrong, see https://app.slack.com/client/T2077MDKQ/C033H6DJSJU

Is it legal to call `torch.compile` several times on the same python object/function?

Reply:
I think if your original graph was a single graph, you don't recompile.

Reply:
You may or may not need to change this behaviour; it's an easy optimization that held previously.

Reply:
> If this PR proposes to capture different graphs based on presence of this function call

Yes, this PR proposes to capture different graphs based on `torch._dynamo.is_fullgraph_compiling()` (i.e. based on `fullgraph` argument). This is working well in case we don't call multiple times `torch.compile` on the same function/object, but currently does not work when calling successively `torch.compile` on the same function/object due to `InstructionTranslator` not being re-initialized.

> then you can make the behaviour stricter - i.e. cause fullgraph flag changing to always recompile - only when your new function call is present; see guards for fullgraph flag.

Could you point me out to the file responsible for that? I could not find `nopython`, `one_graph`, `fullgraph` references in `guards.py` or `mutation_guard.py`

Reply:
It may have been removed unfortunately, a bunch of code got reverted due to some meta internal failures (the investigation hasn't concluded afaik).

You may have to introduce a new guard that recompiles when the fullgraph flag changes, if this behaviour is strictly necessary.

Reply:
It was removed in https://github.com/pytorch/pytorch/pull/115384